app/models/user.rb

```diff
@@ -0,0 +1,47 @@
+class User < ApplicationRecord
+  has_secure_password
+  has_many :products, dependent: :destroy
+
+  attr_accessor :activation_token
+
+  before_create :create_activation_digest
```

Comment:
Vào dự án thực tế, thông thường ae dev sẽ ít dùng callback. Về lâu dài khi code phìng to ra sẽ khó maintain